src/util.js

```diff
@@ -87,12 +90,45 @@ export function serializeForm(form) {
 				values = [values];
 			}
 			values.forEach(value => {
-				let param = [name, value].map(encodeURIComponent).join("=");
-				params.push(param);
+				params.setParameter(name, value);
 			});
 		}
 		return params;
-	}, []).join("&");
+	}, params).serialize();
+}
+
+class FormDataParams {
+	constructor() {
+		this.formData = new FormData();
+	}
+
+	setParameter(name, value) {
+		this.formData.append(name, value);
+	}
+
+	serialize() {
+		return this.formData;
+	}
+}
+
+class StringParams {
+	constructor() {
+		this.params = [];
+	}
+
+	setParameter(name, value) {
+		if(typeof (value) !== "string") {
```

Comment:
when might this occur?

Reply:
Probably a leaky abstraction 🙈 

This occurs if the user calls `submitForm` on a `form` element which does not have the correct `enctype` set. Essentially, when using the `StringParams` accumulator you need to make sure that all of your input parameters can actually be persisted as strings.

Reply:
uhm... this makes me realize that perhaps `serializeForm` isn't even warranted when using `FormData`: doesn't `FormData` itself already provide that very functionality? i.e. [`new FormData(formNode)` should suffice](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData)

Reply:
It's possible. that would be nice.

Reply:
I'm pretty sure that's how it was designed - to be submitted like this:

```javascript
fetch(uri, {
    method: "post",
    body: new FormData(formNode),
    …
}
```

which of course means that `FormData` only makes sense if native `fetch` is available (though that should always be the case IIRC)

either way, we should be explicit about what the progressive-enhancement story here is: at what point do we abort and just let native (i.e. non-AJAX) form submissions kick in? is there anything in between, like `fetch`/`FormData` being unavailable (think IE 11) - in which case stringification would still be an option, files excluded

in other words: for anything other than IE 11, most of the code in this module is unnecessary

Reply:
https://caniuse.com/?search=formdata

FormData is supported on IE11 (at least it says that it is)

Reply:
I've just tested the implementation in IE11 that I am currently using (which uses `FormData` when the `enctype` is correct) and in that case, the upload with ajax works, so I would say that we can probably safely use `FormData`.